week2/index.js

```diff
@@ -1,65 +1,131 @@
+//Which we used it above To assign the variable we will use it later
+function openFile(fileName) {
+	try {
+		var fileContaint = fs.readFileSync(__dirname + '/' + fileName, 'utf8');
+	} catch (error) {
+		if (error.code === 'ENOENT') {
+			return console.log(fileName + ' Not exist! (or your ' + fileName + ' file)');
+		} else {
+			return console.log('Error: Something went wrong', error);
+		}
+	}
+	return fileContaint;
+}
+// To check the type of command that will be typed by the user
 switch (command) {
-  case 'help':
-  default:
-    showHelp();
-    break;
-  case 'list':
-    listTodos();
-    break;
+	case 'help':
+	default:
+		showHelp();
+		break;
+	case 'add':
+		addItem();
+		break;
+	case 'remove':
+		deleteLine();
+		break;
+	case 'reset':
+		resetFile();
+		break;
+	case 'update':
+		updateLine();
+		break;
+	case 'list':
+		listTodos();
+		break;
 }
-
+//Converts the text file into an array to deal with
 function splitStringByNewline(string) {
-  return string.split('\n').filter(function(element) {
-    element = element.trim();
-    return element.length > 0;
-  });
+	return string.split('\n').filter(function (element) {
+		element = element.trim();
+		return element.length > 0;
+	});
 }
 
+//To call the helper text file.
 function showHelp() {
-  openFile('help.txt', function(error, data) {
-    if (error) {
-      return console.log('Error: the help file could not be displayed', error);
-    }
-    console.log(data);
-  });
+	let dataOfHelp = openFile('help.txt')
+	console.log(dataOfHelp);
 }
-
-function listTodos() {
-  openFile('todo.txt', function(error, data) {
-    if (error) {
-      if (error.code === 'ENOENT') {
-        return console.log('Nothing to do! (or your dog ate your todo list)');
-      } else {
-        return console.log('Error: Something went wrong', error);
-      }
-    }
-
-    var todos = splitStringByNewline(data);
-
-    if (todos.length === 0) {
-      return console.log('Nothing to do!')
-    }
-
-    console.log('Your todo list looks like this');
-    todos.forEach(function(element, index) {
-      index = (index + 1).toString();
-      console.log(index, element);
-    });
-
-    if (todos.length > 5) {
-      console.log('You have too much to do!');
-    }
-  });
+//To add a new item.
+function addItem() {
+	//remove first item from process.argv
+	options.shift(); 
+	//Make it on a single line
+	let newItem = options.join(' ') ;
+	// append the new item to list
+	fs.appendFileSync('todo.txt', '\n' + newItem + '\n', 'utf8');
+	//Open the file todo txt and read it again To update the information
+	dataOfTodos = openFile('todo.txt');
+	todos = splitStringByNewline(dataOfTodos);
+	listTodos(); 
 }
-
-function openFile(fileName, callback) {
-  fs.readFile(__dirname + '/' + fileName, 'utf8', function(error, data) {
-    callback(error, data)
-  });
+//To delete the selected item.
+function deleteLine() {
+	//Check for the item number
+	if ( valOfOptions > todos.length || valOfOptions <= 0) { 
```

Comment:
We always preach that whitespace is cheap, but if you're going to use the leading space _inside_ the parentheses like you've done here, you should add the trailing space as well: `if ( valOfOptions > todos.length || valOfOptions <= 0 )`

It's a question of style, and I prefer `if (valOfOptions > todos.length || valOfOptions <= 0)`, but the most important thing is to be consistent.